app/objects/c_operation.py

```diff
@@ -445,21 +448,48 @@ def _check_reason_skipped(self, agent, ability, op_facts, state, agent_executors
             facts = re.findall(BasePlanningService.re_variable, executor.test) if executor.command else []
             if not facts or all(fact in op_facts for fact in facts):
                 fact_dependency_fulfilled = True
-
-        if not agent.trusted:
-            return dict(reason='Agent untrusted', reason_id=self.Reason.UNTRUSTED.value,
+        untrusted_agent = agent.paw in self.untrusted_agents
+        associated_links = set([link.id for link in self.chain if link.paw == agent.paw
+                                and link.ability.ability_id == ability.ability_id])
+        reason_description = 'Untrusted'
+
+        if agent.platform == 'unknown':
```

Comment:
we also want to check whether or not the agent's platform is included in the ability. For instance, if the ability is only for mac/linux, and the agent is running on windows

Reply:
resolving because the next conditional (`elif not valid_executors`) makes this check